### DISCUSSION TITLE
Support CEL language to filter incoming webhook in receivers

### DISCUSSION THREAD

**Discussion by sestegra** [fluxcd/flux2#4393] [Proposals]
Hello,

After reading this issue https://github.com/fluxcd/notification-controller/issues/491, I would like to raise an idea to support more filtering options using the [CEL language](https://github.com/google/cel-spec).

For instance, Tekton triggers supports CEL language.
If you want to test it locally, please have a look at `cel-eval` command
https://github.com/tektoncd/triggers/tree/main/cmd/cel-eval

Would it be feasible to incorporate a logic akin to this in the notification-controller?

**Comment by stefanprodan** [comment]
Sounds like a good idea, we plan to use CEL in other parts of Flux.
